From 7052a12f7e48da510c545446f4b69fa0422b6c49 Mon Sep 17 00:00:00 2001
From: Eddie Huang <eddie.huang@mediatek.com>
Date: Mon, 2 Feb 2015 19:38:08 +0800
Subject: [PATCH 5/5] Fix: arm64: dts: mediatek: Add MT8173 MMC dts

Add node mmc0 and mmc1

Fix clk and pinctrl to new binding

Change-Id: I5104e987443e20f4fa97c0deeb1a62a1650aac07
Signed-off-by: Eddie Huang <eddie.huang@mediatek.com>
---
 arch/arm64/boot/dts/mediatek/mt8173-evb.dts | 16 ++++++++--------
 arch/arm64/boot/dts/mediatek/mt8173.dtsi    |  4 ++--
 2 files changed, 10 insertions(+), 10 deletions(-)

diff --git a/arch/arm64/boot/dts/mediatek/mt8173-evb.dts b/arch/arm64/boot/dts/mediatek/mt8173-evb.dts
index 2050011..7a98fcd 100644
--- a/arch/arm64/boot/dts/mediatek/mt8173-evb.dts
+++ b/arch/arm64/boot/dts/mediatek/mt8173-evb.dts
@@ -238,7 +238,7 @@
 &pio {
 	mmc0_pins_default: mmc0default {
 		pins_cmd_dat {
-			pins = <MT8173_PIN_64_MSDC0_DAT7__FUNC_MSDC0_DAT7>,
+			pinmux = <MT8173_PIN_64_MSDC0_DAT7__FUNC_MSDC0_DAT7>,
 				<MT8173_PIN_63_MSDC0_DAT6__FUNC_MSDC0_DAT6>,
 				<MT8173_PIN_62_MSDC0_DAT5__FUNC_MSDC0_DAT5>,
 				<MT8173_PIN_61_MSDC0_DAT4__FUNC_MSDC0_DAT4>,
@@ -251,14 +251,14 @@
 				bias-pull-up;
 		};
 		pins_clk {
-			pins =  <MT8173_PIN_65_MSDC0_CLK__FUNC_MSDC0_CLK>;
+			pinmux =  <MT8173_PIN_65_MSDC0_CLK__FUNC_MSDC0_CLK>;
 				bias-pull-down;
 		};
 	};
 
 	mmc1_pins_default: mmc1default {
 		pins_cmd_dat {
-			pins = <MT8173_PIN_73_MSDC1_DAT0__FUNC_MSDC1_DAT0>,
+			pinmux = <MT8173_PIN_73_MSDC1_DAT0__FUNC_MSDC1_DAT0>,
 			     <MT8173_PIN_74_MSDC1_DAT1__FUNC_MSDC1_DAT1>,
 			     <MT8173_PIN_78_MSDC1_CMD__FUNC_MSDC1_CMD>,
 			     <MT8173_PIN_75_MSDC1_DAT2__FUNC_MSDC1_DAT2>,
@@ -269,20 +269,20 @@
 		};
 
 		pins_clk {
-			pins = <MT8173_PIN_77_MSDC1_CLK__FUNC_MSDC1_CLK>;
+			pinmux = <MT8173_PIN_77_MSDC1_CLK__FUNC_MSDC1_CLK>;
 			bias-pull-down;
 			drive-strength = <MTK_DRIVE_4mA>;
 		};
 
 		pins_insert {
-			pins= <MT8173_PIN_132_I2S0_DATA1__FUNC_GPIO132>;
+			pinmux = <MT8173_PIN_132_I2S0_DATA1__FUNC_GPIO132>;
 			bias-pull-up;
 		};
 	};
 
 	mmc0_pins_uhs: mmc0@0{
 		pins_cmd_dat {
-			pins = <MT8173_PIN_64_MSDC0_DAT7__FUNC_MSDC0_DAT7>,
+			pinmux = <MT8173_PIN_64_MSDC0_DAT7__FUNC_MSDC0_DAT7>,
 			     <MT8173_PIN_63_MSDC0_DAT6__FUNC_MSDC0_DAT6>,
 			     <MT8173_PIN_62_MSDC0_DAT5__FUNC_MSDC0_DAT5>,
 			     <MT8173_PIN_61_MSDC0_DAT4__FUNC_MSDC0_DAT4>,
@@ -306,7 +306,7 @@
 
 	mmc1_pins_uhs: mmc1@0 {
 		pins_cmd_dat {
-			pins = <MT8173_PIN_73_MSDC1_DAT0__FUNC_MSDC1_DAT0>,
+			pinmux = <MT8173_PIN_73_MSDC1_DAT0__FUNC_MSDC1_DAT0>,
 			     <MT8173_PIN_74_MSDC1_DAT1__FUNC_MSDC1_DAT1>,
 			     <MT8173_PIN_78_MSDC1_CMD__FUNC_MSDC1_CMD>,
 			     <MT8173_PIN_75_MSDC1_DAT2__FUNC_MSDC1_DAT2>,
@@ -317,7 +317,7 @@
 		};
 
 		pins_clk {
-			pins = <MT8173_PIN_77_MSDC1_CLK__FUNC_MSDC1_CLK>;
+			pinmux = <MT8173_PIN_77_MSDC1_CLK__FUNC_MSDC1_CLK>;
 			drive-strength = <MTK_DRIVE_4mA>;
 			bias-pull-down = <MTK_PUPD_SET_R1R0_10>;
 		};
diff --git a/arch/arm64/boot/dts/mediatek/mt8173.dtsi b/arch/arm64/boot/dts/mediatek/mt8173.dtsi
index 4c914f6..8a1a14c 100644
--- a/arch/arm64/boot/dts/mediatek/mt8173.dtsi
+++ b/arch/arm64/boot/dts/mediatek/mt8173.dtsi
@@ -237,7 +237,7 @@
 			compatible = "mediatek,mt8173-mmc","mediatek,mt8135-mmc";
 			reg = <0 0x11230000 0 0x108>;
 			interrupts = <0 71 IRQ_TYPE_LEVEL_LOW>;
-			clocks = <&perisys PERI_MSDC30_0>;
+			clocks = <&pericfg PERI_MSDC30_0>;
 			clock-names = "source";
 			status = "disabled";
 		};
@@ -246,7 +246,7 @@
 			compatible = "mediatek,mt8173-mmc","mediatek,mt8135-mmc";
 			reg = <0 0x11240000 0 0x108>;
 			interrupts = <0 72 IRQ_TYPE_LEVEL_LOW>;
-			clocks = <&perisys PERI_MSDC30_1>;
+			clocks = <&pericfg PERI_MSDC30_1>;
 			clock-names = "source";
 			status = "disabled";
 		};
-- 
1.9.1

